src/parse.ts

```diff
@@ -76,12 +80,21 @@ export function parse(messageDefinition: string, options: ParseOptions = {}): Me
       : buildType(definitionLines, ROS1_GRAMMAR),
   );
 
+  // Filter out duplicate types.
+  // This will avoid that searching a type by name will return more than one result.
+  const seenTypes: MessageDefinition[] = [];
+  const uniqueTypes = types.filter((definition) => {
+    return seenTypes.find((otherDefinition) => isMsgDefEqual(definition, otherDefinition))
```

Comment:
> is this deduplication as performant as it could be or as performant as `_.uniqBy`?

Probably not. OK to add lodash as a dependency? If so then I'll change it to use `_.uniqBy

Reply:
Can you please not add a dependency to this project for a bug fix?

Reply:
Yeah I'd avoid adding the dep. Just wanted to see if we could get this down to a single iteration through the types, but it's not strictly necessary